File: BLL/MapModels/CommentModel.cs
namespace Business.MapModels
{
    /// <summary>
    /// Represents Comment mapped model.
    /// </summary>
    public class CommentModel : BaseModel
    {
        /// <summary>
        /// Comment's text.
        /// </summary>
        public string Text { get; set; }

        /// <summary>
        /// Comment's rating.
        /// </summary>
        public int Rating { get; set; }

        /// <summary>
        /// Comment's author's id.
        /// </summary>
        public int AuthorId { get; set; }

        /// <summary>
        /// Comment's author's name.
        /// </summary>
        public string AuthorName { get; set; }

        /// <summary>
        /// Comment's posting date.
        /// </summary>
        public string PostingDate { get; set; }

        /// <summary>
        /// Comment's topic's id..
        /// </summary>
        public int TopicId { get; set; }

        /// <summary>
        /// Comment's topic's name.
        /// </summary>
        public string TopicName { get; set; }

        /// <summary>
        /// Comment's community's name.
        /// </summary>
        public string CommunityName { get; set; }
    }
}

File: Forum.WebApi/ForumAngularClient/src/app/models/comment/comment.model.ts
export class CommentModel {
  constructor(
    public id?: number,
    public text?: string,
    public rating?: number,
    public authorId?: number,
    public postingDate?: string,
    public authorName?: string,
    public topicId?: number,
    public topicName?: string,
    public communityName?: string,
    public isEditing: boolean = false,
    public isRatedUp: boolean = false,
    public isRatedDown: boolean = false
  ) {
  }
}
